src/MoBi.Presentation/Presenter/EditIndividualAndExpressionConfigurationsPresenter.cs

```diff
@@ -97,14 +106,28 @@ public void RemoveSelectedExpression(ITreeNode selectedNode)
          _view.RemoveUsedExpression(selectedNode);
       }
 
-      public void AddSelectedExpression(ITreeNode selectedNode)
+      public void AddSelectedExpressions(IReadOnlyList<ITreeNode> selectedNodes)
+      {
+         // We need the ToList because all nodes must be evaluated, then we are testing if any
+         // nodes failed to be added to the selection
+         var nodesNotAdded = selectedNodes.Where(x => !addSelectedExpression(x)).Select(x => x.TagAsObject as ExpressionProfileBuildingBlock).ToList();
+         if (nodesNotAdded.Any())
+            _dialogCreator.MessageBoxError(AppConstants.Captions.CouldNotAddExpressionsDuplicatingMolecule(nodesNotAdded.AllNames()));
+      }
+
+      private bool addSelectedExpression(ITreeNode selectedNode)
       {
          if (!(selectedNode.TagAsObject is ExpressionProfileBuildingBlock expression))
```

Comment:
this is done so many time in this file (already three time), maybe a method to get the expression from a node?